test/Microsoft.AspNet.Mvc.TagHelpers.Test/Internal/ModeMatchResultTest.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Logging;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    public class ModeMatchResultTest
+    {
+        [Fact]
+        public void LogDetails_LogsVerboseWhenNoFullMatchesFound()
+        {
+            // Arrange
+            var modeMatchResult = new ModeMatchResult<string>();
+            var logger = MakeLogger(LogLevel.Verbose);
+            var tagHelper = new Mock<ITagHelper>();
+            var uniqueId = "id";
+
+            // Act
+            modeMatchResult.LogDetails(logger, tagHelper.Object, uniqueId);
+
+            // Assert
+            Mock.Get(logger).Verify(l => l.Write(
```

Comment:
I'm guessing you mean make a test logger that makes the messages available for verification? That's doable, but I personally find this much more maintainable. We can debate in person if you like :smile: